MockDuck/Sources/MockBundle.swift

```diff
@@ -113,14 +109,14 @@ final class MockBundle {
 
                 // write out request body if the format is supported.
                 // This should be the same filename with a different extension.
-                if let requestBodyFileName = SerializationUtils.fileName(for: .requestBody(requestResponse)) {
+                if let requestBodyFileName = requestResponse.fileName(for: .requestBody) {
                     let requestBodyURL = recordingURL.appendingPathComponent(requestBodyFileName)
                     try requestResponse.request.httpBody?.write(to: requestBodyURL, options: [.atomic])
```

Comment:
Some times the baby of a request is inside `request.httpBodyStream` not `request.httpBody`, so when the mime type of request is `application/json` and `request.httpBody` is empty writing of the '-request' file will fail here, but the library will try to read that file when loading the request.

Reply:
We added an extension on URLRequest to also look at that property when the body is nil.

```swift
internal extension URLRequest {

    var bodyStreamData: Data? {

        guard let bodyStream = self.httpBodyStream else { return nil }

        bodyStream.open()

        // Will read 16 chars per iteration. Can use bigger buffer if needed
        let bufferSize: Int = 16

        let buffer = UnsafeMutablePointer<UInt8>.allocate(capacity: bufferSize)

        var dat = Data()

        while bodyStream.hasBytesAvailable {

            let readDat = bodyStream.read(buffer, maxLength: bufferSize)
            dat.append(buffer, count: readDat)
        }

        buffer.deallocate()

        bodyStream.close()

        return dat
    }
}

```

We are using it in:
`record` method

```swift
let body = requestResponse.request.httpBody ?? requestResponse.request.bodyStreamData
try body?.write(to: requestBodyURL, options: [.atomic])
```

instead of:

```swift
try requestResponse.request.httpBody?.write(to: requestBodyURL, options: [.atomic])
```

and 

`serializedHashValue`

```swift
if let body = normalizedRequest.httpBody ?? normalizedRequest.bodyStreamData {
    if let json = try? JSONSerialization.jsonObject(with: body, options: []) as! [String: Any] {
             hashData.append(sort(dictionary: json).data(using: .utf8)!)
    } else {
        hashData.append(body)
    }
}
```

instead of 

```swift

if let body = normalizedRequest.httpBody {
     hashData.append(body)
}

```

Reply:
I like this change, too, but would prefer for it to be done in its own PR.

Reply:
See #24

Reply:
I will open a separate PR for it after you merge this one

Reply:
https://github.com/buzzfeed/MockDuck/pull/28